conflictlog/record.go

```diff
@@ -127,18 +179,35 @@ func (r *ConflictRecord) SmallString() string {
 		r.Source.Scope, r.Source.Collection, r.Target.Scope, r.Target.Collection)
 }
 
-// populates some derived data.
-func (r *ConflictRecord) PopulateData(replicationId string) {
+// populates the following:
+// 1. document Ids.
+// 2. CRD body.
+// 3. xattr for all three conflict records.
+func (r *ConflictRecord) PopulateData(replicationId string) error {
 	if r == nil {
-		return
+		return fmt.Errorf("nil CRD")
 	}
 
 	now := time.Now().UnixNano()
 
 	r.ReplicationId = replicationId
+
 	r.PopulateSourceDocId(now)
 	r.PopulateTargetDocId(now)
 	r.PopulateCRDocId(now)
+
+	body, err := json.Marshal(r)
+	if err != nil {
```

Comment:
Add context for the error before the generic marshal error - could be hard to find where the issue is

Reply:
Done